code/include/playfab/internal_memory.h

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+#include <set>
+#include <queue>
+#include <unordered_set>
+
+namespace PlayFab
+{
+
+//------------------------------------------------------------------------------
+// Allocator
+//------------------------------------------------------------------------------
+template<class T>
+struct Allocator
+{
+    using value_type = T;
+    using is_always_equal = std::true_type;
+
+    Allocator() = default;
+    template<class U>
+    Allocator(Allocator<U> const&) {}
+
+    T* allocate(size_t n);
+    void deallocate(_In_opt_ void* p, size_t) noexcept;
+};
+
+template<class T>
+bool operator==(Allocator<T> const&, Allocator<T> const&);
+template<class T>
+bool operator!=(Allocator<T> const&, Allocator<T> const&);
+
+//------------------------------------------------------------------------------
+// Deleter
+//------------------------------------------------------------------------------
+template<class T>
+struct Deleter
+{
+    void operator()(T* ptr) const noexcept;
+};
+
+//------------------------------------------------------------------------------
+// Smart Pointers
+//------------------------------------------------------------------------------
+template<class T>
+using SharedPtr = std::shared_ptr<T>;

```

Comment:
I think I would disagree with many of these simple-rename aliases, as the syntactical reduction is trivial, and it obscures what's actually being used. As somebody who wants to learn what's really happening when I read this code, these simple rename aliases make it harder for me to do that. The clarity and learning cost is too high compared to just what the syntax-prettiness gives you.

Reply:
I'll be honest, I pulled this file directly from XSAPI's codebase and I don't have a super strong opinion here.  I can see the point of view that adding this indirection is harmful, but I think the UniquePtr alias is cleaner then specifying the custom deleter everywhere. Given that, I think aliasing all the pointer types to keep things consistent would be my preference.

---
In reply to: [590890227](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590890227) [](ancestors = 590890227)